### PR TITLE
#109: Gallery deletion crash

### PR DESCRIPTION
**Pull request by mdrlzy**
#109

### CODE REVIEW

File: app/src/main/java/space/taran/arknavigator/mvp/presenter/adapter/ResourcesGridPresenter.kt [hunk 1]
```diff
@@ -51,6 +53,9 @@ class ResourcesGridPresenter(
     fun getCount() = selection.size
 
     fun bindView(view: FileItemView) {
+        if (!resourcesReady)
```
Comment (darkovski88) [inline app/src/main/java/space/taran/arknavigator/mvp/presenter/adapter/ResourcesGridPresenter.kt:56]: is it a better idea to have this isResourceReady to be dependent on something else - not based on which method from the presenter is actually being called. For example to be dependant of some state(if something has finished or not)?

File: app/src/main/java/space/taran/arknavigator/mvp/presenter/adapter/ResourcesGridPresenter.kt [hunk 2]
```diff
@@ -79,6 +84,7 @@ class ResourcesGridPresenter(
 
     suspend fun updateSelection(selection: Set<ResourceId>) = withContext(Dispatchers.Default) {
         this@ResourcesGridPresenter.selection = resources.filter { selection.contains(it.id) }
```
Comment (darkovski88) [inline app/src/main/java/space/taran/arknavigator/mvp/presenter/adapter/ResourcesGridPresenter.kt:86]: maybe it's a better idea to have this filtration of resources to be taken care from another class which will serve as a filter for this kind of things. Basically the job of the presenter is to do the logic(not to be done on the view). Also if we extract these kind of logics to separate classes we will have cleaner code, separation of concerns, and unit tests will be more easy to do.

Comment (kirillt) [reply app/src/main/java/space/taran/arknavigator/mvp/presenter/adapter/ResourcesGridPresenter.kt:86]: @darkovski88 actually we have such an entity, _tags selector_. Classes `TagsSelectorPresenter` and `TagsSelectorAdapter`. Maybe the architecture can be improved here.

Comment (darkovski88) [reply app/src/main/java/space/taran/arknavigator/mvp/presenter/adapter/ResourcesGridPresenter.kt:84]: @kirillt as i can see here the code and app looks ok to me. I have no negative comments on this PR except maybe take some action to the ones that i've already added

**Review by darkovski88 — APPROVED** [review]
Looks OK to me..

**Comment by kirillt** [comment]
Substituted with https://github.com/ARK-Builders/ARK-Navigator/pull/172

**Pull request closed without merging**